### ISSUE TITLE
ESLint formatter can't distinguish between an internal error and a misconfiguration

### ISSUE THREAD

**Issue by lgolding** [microsoft/sarif-sdk#1798]
Certain ESLint "misconfigurations" -- such as running ESLint on a DLL instead of on a JS or TS file, or attempting to analyze a dialect for which the ESLint configuration file is not appropriate -- result in ESLint messages with no ruleId, and message text starting with "Parsing error:". At present, we place _all_ ESLint messages with no ruleId into `invocation.toolConfigurationNotifications`. But it's possible that some of them would be better placed in `toolExecutionNotifications`.

We should investigate whether there is a way to distinguish user configuration errors from tool execution errors, for example by examining the ESLint source code or engaging with the ESLint team.

For now we place them in `toolConfigurationErrors` for the reason explained by @michaelcfanning. We have a choice between

1. A very small number of tool owners receive problem reports for every user’s misconfiguration plus all valid catastrophic engine exceptions OR
2. A larger number of tool users receive problem reports for their misconfigurations and the typically rare catastrophic engine issues they happen to provoke.

Option <span>#</span>2 will cause the least amount of damage for now.

**Comment by michaelcfanning** [comment]
Merged.

**Comment by lgolding** [comment]
Right, this is still open because although the SARIF is now valid, we'd still like to distinguish these cases so we can route problems to the proper audience (tool developer or build engineer).
